### DISCUSSION TITLE
Is there a way to use Pyarmor without the internet for Python 3.11 and above?

### DISCUSSION THREAD

**Discussion by khadeom** [dashingsoft/pyarmor#1671] [Q&A]
Hi Pyarmor Team,

I have been using the commercial version of Pyarmor for a few years now & I am happy with it. 

I am looking to upgrade to the latest version as I would like to enjoy the benefits of Python 3.12 and beyond with Pyarmor.  I am looking for a version of Pyarmor which works with Python 3.12 but does not require an internet connection, like my current Pyarmor version. I am willing to buy/upgrade to Pyarmor 8+ to use these features which satisfy my requirements.

Look forward to a reply.

Thanks & Regards,
Om Khade

[ This question was initially posed via email, but we were instructed to submit it on GitHub. ]

**Comment by jondy** [comment]
It need Pyarmor Group License, more information please check latest pyarmor documentation

**Comment by khadeom** [comment]
Thanks @jondy. I've decided to proceed with the PyArmor group license. Could you please clarify if it allows me to share the obfuscated code with my friend? Additionally, will he be able to run the obfuscated code on his machine, even if it's different from the one where the code was originally obfuscated using PyArmor? Your assistance is highly valued.

**Comment by jondy** [comment]
You can distribute the obfuscated scripts to anyone, but if target machine platform is different from build machine, please use the right option `--platform` to obfuscate the scripts.
